### PR TITLE
Code generation for MutableUiState

### LINKED ISSUE #405: Code generation for MutableUiState
`MutableUiState` instances contain repetitive code and could be generated based on the corresponding `TargetUiState` class. It would make it a lot easier and quicker to implement ui representations.

### PR DESCRIPTION
**Pull request by manuel-martos**
## Description

...

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: appyx-components/internal/common/build.gradle.kts [hunk 1]
```diff
@@ -33,7 +37,7 @@ kotlin {
         }
         val androidMain by getting
         val desktopMain by getting
-        val jsMain by getting
+//        val jsMain by getting
```
Comment (zsoltk) [inline appyx-components/internal/common/build.gradle.kts]: any reason behind this, or is it just an accident?

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/MutableUiStateSpecs.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.SOURCE)
+@Suppress("Unused")
+annotation class MutableUiStateSpecs(
+    val playMode: PlayMode = PlayMode.ConcurrentMode
+) {
+    enum class PlayMode {
+        SequentialMode, ConcurrentMode
+    }
```
Comment (zsoltk) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/MutableUiStateSpecs.kt]: Maybe I'd rename to:

```suggestion
    enum class AnimationMode {
        SEQUENTIAL, CONCURRENT
    }
```

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/MutableUiStateSpecs.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.bumble.appyx.interactions.core.ui.state
```
Comment (zsoltk) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/MutableUiStateSpecs.kt:1]: Should this annotation live in the processor's module or is it better here?

**Review by zsoltk — APPROVED** [review]

File: appyx-components/internal/common/build.gradle.kts (hunk 1 above)
Comment (manuel-martos) [reply appyx-components/internal/common/build.gradle.kts]: Good catch. It's not harmful but I'll uncomment it.

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/MutableUiStateSpecs.kt (hunk 3 above)
Comment (manuel-martos) [reply appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/MutableUiStateSpecs.kt:1]: All samples I've been analysing of KSP processors have the annotation out of the processor module.

**Review by zsoltk — APPROVED** [review]

**Pull request merged**